x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/Param.java

```diff
@@ -11,18 +11,20 @@
 /**
  * Represent a strongly typed parameter value
  */
-public class TypedParamValue {
+public class Param {
```

Comment:
I guess this was kept around to still support the `type`, right? Otherwise Params could probably do what we need.

Reply:
I'd like to keep Param, we can store the inferred DataType in it, and if the actual value is a null, we can have a Param representation of null.